### ISSUE TITLE
💅 False positive for "is defined but never used"

### ISSUE THREAD

**Issue by tolokoban** [biomejs/biome#9008]
### Environment information

It seems that deconstructing is not well managed.

In the following example, `myVar` is used in `print()` like this: `const { myVar } = this;`.
But Biome still complains.

<details>
<summary>Code example</summary>

```ts
export class MyClass {
  private readonly myVar: string;

  constructor() {
    this.myVar = "Hello";
  }

  print() {
    const { myVar } = this;
    console.log(myVar);
  }
}
```

</details>

### Rule name

noUnusedPrivateClassMembers

### Playground link

https://biomejs.dev/playground/?code=ZQB4AHAAbwByAHQAIABjAGwAYQBzAHMAIABNAHkAQwBsAGEAcwBzACAAewAKACAAIABwAHIAaQB2AGEAdABlACAAcgBlAGEAZABvAG4AbAB5ACAAbQB5AFYAYQByADEAOgAgAHMAdAByAGkAbgBnADsACgAgACAAcAByAGkAdgBhAHQAZQAgAHIAZQBhAGQAbwBuAGwAeQAgAG0AeQBWAGEAcgAyADoAIABzAHQAcgBpAG4AZwA7AAoACgAgACAAYwBvAG4AcwB0AHIAdQBjAHQAbwByACgAKQAgAHsACgAgACAAIAAgAHQAaABpAHMALgBtAHkAVgBhAHIAMQAgAD0AIAAiAEgAZQBsAGwAbwAiADsACgAgACAAIAAgAHQAaABpAHMALgBtAHkAVgBhAHIAMgAgAD0AIAAiAFcAbwByAGwAZAAiADsACgAgACAAfQAKAAoAIAAgAHAAcgBpAG4AdAAoACkAIAB7AAoAIAAgACAAIABjAG8AbgBzAHQAIAB7ACAAbQB5AFYAYQByADEAIAB9ACAAPQAgAHQAaABpAHMAOwAKACAAIAAgACAAYwBvAG4AcwBvAGwAZQAuAGwAbwBnACgAbQB5AFYAYQByADEAKQA7AAoAIAAgACAAIABjAG8AbgBzAG8AbABlAC4AbABvAGcAKAB0AGgAaQBzAC4AbQB5AFYAYQByADIAKQAKACAAIAB9AAoAfQA%3D

### Expected result

I would expect that both direct access and destructuring work the same.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by solssak** [comment]
I'd like to work on this

**Comment by solssak** [comment]
I opened a PR to fix this: #9047

**Comment by ematipico** [comment]
closing, this is now fixed.
